test/dst/generator.go

```diff
@@ -193,62 +194,19 @@ func (g *Generator) GenerateCreatePromise(r *rand.Rand, t int64) *t_api.Request
 	}
 }
 
-func (g *Generator) GenerateCancelPromise(r *rand.Rand, t int64) *t_api.Request {
+func (g *Generator) GenerateCompletePromise(r *rand.Rand, t int64) *t_api.Request {
 	id := g.idSet[r.Intn(len(g.idSet))]
 	idempotencyKey := g.idemotencyKeySet[r.Intn(len(g.idemotencyKeySet))]
 	data := g.dataSet[r.Intn(len(g.dataSet))]
 	headers := g.headersSet[r.Intn(len(g.headersSet))]
 	strict := r.Intn(2) == 0
+	state := promise.State(math.Exp2(float64(r.Intn(4) + 1)))
```

Comment:
Oh nice this is very clever! The current state enum looks like this
```go
const (
	Pending State = 1 << iota
	Resolved // 2
	Rejected // 4
	Timedout // 8
	Canceled // 16
)
```

I recommend swapping Timedout and Canceled so that the 3 valid states for the `CompletePromise` request will be in order. That way we can still use this random approach, also it feels better to me have Timedout last for some reason - what do you think?

Reply:
Yes, I think that order makes more sense. 

Working on it.

Reply:
Done, thanks!